### PR TITLE
Alee/cvsl 1910 c

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: server/@types/managedCase.ts [hunk 1]
```diff
@@ -48,11 +48,10 @@ export type ApprovalLicence = {
   isDueToBeReleasedInTheNextTwoWorkingDays: boolean
 }
 
-export type GenericManagedCase<T> = {
+export type ManagedCase = {
   deliusRecord?: DeliusRecord
   nomisRecord?: CvlPrisoner
-  licences?: T[]
+  licences?: Licence[]
```
Comment (andrewrlee) [inline server/@types/managedCase.ts:54]: doesn't need to be generic

File: server/routes/viewingLicences/handlers/viewCases.ts [hunk 2]
```diff
@@ -87,29 +78,27 @@ export default class ViewAndPrintCaseRoutes {
...
-    const casesToView = view === 'prison' ? caselist.getPrisonView() : caselist.getProbationView()
+    const casesToView =
+      view === 'prison'
+        ? await this.caseloadService.getPrisonView(user, prisonCaseloadToDisplay)
+        : await this.caseloadService.getProbationView(user, prisonCaseloadToDisplay)
 
     const caseloadViewModel = casesToView.map(c => {
-      let latestLicence = _.head(c.licences)
-      if (!probationView && c.licences.length > 1) {
-        latestLicence = this.findLatestLicence(c.licences)
```
Comment (andrewrlee) [inline server/routes/viewingLicences/handlers/viewCases.ts:96]: pushed down into caseload service

File: server/routes/viewingLicences/handlers/viewCases.ts [hunk 3]
```diff
@@ -144,17 +133,4 @@ export default class ViewAndPrintCaseRoutes {
       probationView,
     })
   }
-
-  GET_WITH_EXCLUSIONS = async (req: Request, res: Response): Promise<void> => {
```
Comment (andrewrlee) [inline server/routes/viewingLicences/handlers/viewCases.ts:148]: not used anymore

File: server/services/licenceService.ts [hunk 4]
```diff
@@ -488,21 +486,6 @@ export default class LicenceService {
     return this.licenceApiClient.batchInActivateLicences(licenceIds)
   }
 
-  public static getLicenceType = (sentenceDetail: CvlPrisoner): LicenceType => {
```
Comment (andrewrlee) [inline server/services/licenceService.ts:491]: this was duplicated - moved to single representation

File: server/services/lists/caCaseloadService.test.ts [hunk 5]
```diff
@@ -800,11 +800,89 @@ describe('Caseload Service', () => {
...
+      const c = createCase(LicenceStatus.NOT_IN_PILOT, pastDate, '2022-10-19')
+      expect(serviceUnderTest.isProbationCase(c)).toBe(false)
+    })
+    it('should return exclusion for out-of-scope status and future CRD', () => {
+      const c = createCase(LicenceStatus.OOS_BOTUS, futureDate, '2022-10-19')
+      expect(serviceUnderTest.isProbationCase(c)).toBe(false)
+    })
+    it('should return exclusion for out-of-scope status and past CRD', () => {
+      const c = createCase(LicenceStatus.OOS_BOTUS, pastDate, '2022-10-19')
+      expect(serviceUnderTest.isProbationCase(c)).toBe(false)
```
Comment (andrewrlee) [inline server/services/lists/caCaseloadService.test.ts:863]: extracted these tests out of omucaselist

File: server/services/lists/caCaseloadService.ts [hunk 6]
```diff
@@ -1,26 +1,89 @@
...
+      licence: this.findLatestLicence(licences),
+    }))
+  }
+
+  public async getProbationView(user: User, prisonCaseload: string[]): Promise<Case[]> {
+    const cases = await this.getOmuCaseload(user, prisonCaseload)
+    return cases.filter(this.isProbationCase).map(({ licences, ...c }) => ({
+      ...c,
+      licence: licences[0],
+    }))
```
Comment (andrewrlee) [inline server/services/lists/caCaseloadService.ts:72]: will move this lower down the pipeline when we have some decent tests here

File: server/services/lists/caCaseloadService.ts [hunk 7]
```diff
@@ -52,7 +115,7 @@ export default class CaCaseloadService {
     const [withLicence, pending] = await Promise.all([casesWithLicences, casesPendingLicence])
     const casesWithComs = await this.mapResponsibleComsToCases(withLicence.concat(pending))
 
-    return new OmuCaselist(casesWithComs)
+    return casesWithComs
```
Comment (andrewrlee) [inline server/services/lists/caCaseloadService.ts:118]: removing wrapper as logic now done in service

File: server/services/lists/promptListService.ts [hunk 8]
```diff
@@ -20,46 +17,6 @@ export default class PromptListService {
     private readonly licenceService: LicenceService
   ) {}
 
-  async getOmuCaseload(user: User, prisonCaseload: string[]): Promise<OmuCaselist> {
```
Comment (andrewrlee) [inline server/services/lists/promptListService.ts:23]: these weren't being using by this new service so removed

**Review by nishanthmahasamudram — APPROVED** [review]
LGTM

**Pull request merged**